src/scheduler/fifo.c

```diff
@@ -1242,17 +1242,12 @@ run_job(int pbs_sd, resource_resv *rjob, char *execvnode, int throughput, schd_e
 					snprintf(logbuf, MAX_LOG_SIZE, "Job will run for duration=%s", timebuf);
 					schdlog(PBSEVENT_SCHED, PBS_EVENTCLASS_JOB, LOG_NOTICE, rjob->name, logbuf);
 				}
-				if (throughput)
-					rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
-				else
-					rc = pbs_runjob(pbs_sd, rjob->name, execvnode, NULL);
 			}
-		} else {
-			if (throughput)
-				rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
-			else
-				rc = pbs_runjob(pbs_sd, rjob->name, execvnode, NULL);
 		}
+		if (rc > 0 && throughput)
+			rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
+		else
```

Comment:
I agree with you! There seems to be a problem when update_job_attr() returns a 0. Well most of the time while running a job when it is called it isn't going to throw 0 as return value but if send_attr_updates returns a 0 (because of alterjob failure) then we will end up assuming that we ran the job when actually we didn't.
It is unfortunate that we lack consistency in return values that our functions return.
But, do you want to address this issue with this IFL refactoring issue? I think it is better to log a separate ticket and address it there. What do you think?